openassessment/data.py

```diff
@@ -333,4 +334,151 @@ def _use_read_replica(self, queryset):
+            returned_string += u"-- scored_at: {}\n".format(assessment.scored_at)
+            returned_string += u"-- type: {}\n".format(assessment.score_type)
+            returned_string += u"-- scorer_id: {}\n".format(assessment.scorer_id)
+            if assessment.feedback != u"":
+                returned_string += u"-- overall_feedback: {}\n".format(assessment.feedback)
+        return returned_string
+
+    @classmethod
+    def _build_assessments_parts_cell(cls, assessments):
+        """
+        Args:
+            assessments (QuerySet) - assessments containing the parts that we would like to collate into one column.
+        Returns:
+            string that should be included in the relevant 'assessments_parts' column for this set of assessments' row
+        """
+        returned_string = u""
+        for assessment in assessments:
+            returned_string += u"Assessment #{}\n".format(assessment.id)
+            for part in assessment.parts.all():
+                returned_string += u"-- {}".format(part.criterion.label)
+                option_label = part.option.label if part.option.label is not None else u""
+                returned_string += u": {option_label} ({option_points})\n".format(
+                    option_label=option_label, option_points=part.option.points
+                )
+                if part.feedback != u"":
+                    returned_string += u"-- feedback: {}\n".format(part.feedback)
+        return returned_string
+
+    @classmethod
+    def _build_feedback_options_cell(cls, assessments):
+        """
+        Args:
+            assessments (QuerySet) - assessment that we would like to use to fetch and read the feedback options.
+        Returns:
+            string that should be included in the relevant 'feedback_options' column for this set of assessments' row
+        """
+
+        returned_string = u""
+        for assessment in assessments:
+            for feedback in assessment.assessment_feedback.all():
+                for option in feedback.options.all():
+                    returned_string += option.text + u"\n"
+
+        return returned_string
+
+    @classmethod
+    def _build_feedback_cell(cls, assessments):
+        """
+        Args:
+            assessments (QuerySet) - assessment that we would like to use to fetch and read the feedback text.
+        Returns:
+            string that should be included in the relevant 'feedback' column
+        """
+        returned_string = u""
+        if assessments.count() > 0 and assessments[0].assessment_feedback.count() > 0:
+            returned_string = assessments[0].assessment_feedback.all()[0].feedback_text
+        return returned_string
+
+    @classmethod
+    def collect_ora2_data(cls, course_id):
+        """
+        Query MySQL database for aggregated ora2 response data.
+        """
+
+        all_submission_information = sub_api.get_all_course_submission_information(course_id, 'openassessment')
```

Comment:
Note for myself so I can find the submissions integration point quickly when I need it again